src/platform_impl/linux/wayland/pointer.rs

```diff
@@ -241,3 +253,13 @@ pub fn implement_relative_pointer<T: 'static>(
         )
     })
 }
+
+pub fn implement_locked_pointer(
+    surface: &WlSurface,
+    pointer: &WlPointer,
+    constraints: &ZwpPointerConstraintsV1,
+) -> Result<ZwpLockedPointerV1, ()> {
+    constraints.lock_pointer(surface, pointer, None, Lifetime::Persistent.to_raw(), |c| {
```

Comment:
This need for `to_raw()` here is a deficiency in the code generated by wayland-protocols, it self caused by deficiency in the protocol XML files :/ I have "fixing this" somewhere in my todo list.